src/main/java/ua/vedroid/cinema/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,69 @@
+package ua.vedroid.cinema.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import ua.vedroid.cinema.dao.RoleDao;
+import ua.vedroid.cinema.exception.DataProcessingException;
+import ua.vedroid.cinema.model.Role;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private static final Logger log = LogManager.getLogger(RoleDaoImpl.class);
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public RoleDaoImpl(SessionFactory sessionFactory) {
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Role add(Role role) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(role);
+            transaction.commit();
+            log.info("Added new Role: " + role);
+            return role;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t insert Role entity " + role, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Role> getAll() {
```

Comment:
doesn't seem that we need this method (it's never used) i suggest we remove it for now